runtime/src/dpe_crypto.rs

```diff
@@ -78,21 +96,55 @@ impl<'a> Crypto for DpeCrypto<'a> {
 
     fn derive_cdi(
         &mut self,
-        _algs: AlgLen,
-        _measurement: &Digest,
-        _info: &[u8],
+        algs: AlgLen,
+        measurement: &Digest,
+        info: &[u8],
     ) -> Result<Self::Cdi, CryptoError> {
-        Err(CryptoError::NotImplemented)
+        match algs {
+            AlgLen::Bit256 => Err(CryptoError::Size),
+            AlgLen::Bit384 => {
+                let mut hmac_key = [0u8; AlgLen::Bit384.size()];
+                hmac_key.copy_from_slice(measurement.bytes());
+                hmac384_kdf(
+                    self.hmac384,
+                    Hmac384Key::Array4x12(&Array4x12::from(hmac_key)),
+                    b"derive_cdi",
+                    Some(info),
+                    &mut self.trng,
+                    KeyWriteArgs::new(KEY_ID_RT_CDI, KeyUsage::default().set_hmac_data_en()).into(),
+                )
+                .map_err(|_| CryptoError::CryptoLibError)?;
+                Ok(KEY_ID_RT_CDI)
+            }
+        }
     }
 
     fn derive_private_key(
         &mut self,
-        _algs: AlgLen,
-        _cdi: &Self::Cdi,
-        _label: &[u8],
-        _info: &[u8],
+        algs: AlgLen,
+        cdi: &Self::Cdi,
+        label: &[u8],
+        info: &[u8],
     ) -> Result<Self::PrivKey, CryptoError> {
-        Err(CryptoError::NotImplemented)
+        match algs {
+            AlgLen::Bit256 => Err(CryptoError::Size),
+            AlgLen::Bit384 => {
+                hmac384_kdf(
+                    self.hmac384,
+                    KeyReadArgs::new(*cdi).into(),
+                    label,
+                    Some(info),
+                    &mut self.trng,
+                    KeyWriteArgs::new(
+                        KEY_ID_RT_PRIV_KEY,
```

Comment:
We'll want to pick another keyvault slot that's not currently in use. It looks like ROM and FMC use up to keyvault slot 5. So we can use 6 & 7 for the CDI and Private Key